opengever/maintenance/scripts/repository_migration.py

```diff
@@ -765,18 +764,34 @@ def validate_operation(self, operation):
                     "\nInvalid operation: blocking inheritance without setting "
                     "local roles. {}\n".format(operation))
                 operation['is_valid'] = False
-            if has_local_roles:
-                obj = unrestrictedUuidToObject(operation['uid'])
-                if obj:
-                    # newly created positions will have the local_roles set
-                    # in the pipeline
-                    operation['set_permissions'] = True
-                if obj and RoleAssignmentManager(obj).get_assignments_by_cause(ASSIGNMENT_VIA_SHARING):
+            obj = unrestrictedUuidToObject(operation['uid'])
+            if not obj:
+                # newly created positions will have the local_roles set
+                # in the pipeline
+                pass
+            elif self.are_permissions_updated(obj, permissions, inheritance_blocked):
+                operation['set_permissions'] = True
+                if RoleAssignmentManager(obj).get_assignments_by_cause(ASSIGNMENT_VIA_SHARING):
                     operation['local_roles_deleted'] = True
                     logger.warning(
                         "\nSharing assignments for {} will be deleted and "
                         "replaced.\n".format(obj.absolute_url_path()))
 
+    def are_permissions_updated(self, obj, permissions, inheritance_blocked):
+        if inheritance_blocked != getattr(obj, '__ac_local_roles_block__', False):
+            return True
+        old_permissions = {role_shortname: [] for role_shortname in managed_roles_shortnames}
+        assignments = RoleAssignmentManager(obj).get_assignments_by_cause(ASSIGNMENT_VIA_SHARING)
+        for assignment in assignments:
+            for role in assignment['roles']:
+                shortname = SHORTNAMES_BY_ROLE[role]
+                old_permissions[shortname].append(assignment['principal'])
```

Comment:
`shortname` can be any role from the assignment. But the `old_permissions` dict only provides `managed_roles_shortnames` which is a subset of `SHORTNAMES_BY_ROLE`. I think this will not work